processing/src/main/java/org/apache/druid/data/input/impl/RegexInputFormat.java

```diff
@@ -96,10 +96,6 @@ public InputEntityReader createReader(InputRowSchema inputRowSchema, InputEntity
   @Override
   public long getWeightedSize(String path, long size)
   {
-    CompressionUtils.Format compressionFormat = CompressionUtils.Format.fromFileName(path);
-    if (CompressionUtils.Format.GZ == compressionFormat) {
-      return size * CompressionUtils.COMPRESSED_TEXT_WEIGHT_FACTOR;
-    }
-    return size;
+    return size * CompressionUtils.estimatedCompressionFactor(CompressionUtils.Format.fromFileName(path));
```

Comment:
Perhaps we can extend `CompressionUtils.estimatedCompressionFactor()` or add a thin wrapper that accepts the `path` directly and returns the compression factor - internally invoking `CompressionUtils.Format.fromFileName()`

Reply:
thanks for reviewing!